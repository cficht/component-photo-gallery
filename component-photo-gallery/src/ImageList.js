import React, { Component } from 'react';
import ImageItem from './ImageItem.js';


export default class ImageList extends Component {
    state = { 
        selected: null
    }

    render() {
        const imageData = this.props.images;
        const filteredData = imageData.filter(horned => {
            if(!this.state.selected ) return true;
                return horned.keyword === this.state.selected || horned.horns === Number(this.state.selected);
        })
        const imageComponents = filteredData.map(image => <ImageItem horned={image} />)
           
        const handleChange = e => {
            this.setState({selected: e.target.value})
        }

        return (
            <div id="list-main">
                <div id="list-top">
                    <select onChange={handleChange}>
                        {this.props.images.map(image => <option>{image.keyword}</option>)}
                    </select>
                    <select onChange={handleChange}>
                        {this.props.images.map(image => <option>{image.horns}</option>)}
                    </select>
                </div>
                <div id="image-list">
                    <ul>
                        {imageComponents}
                    </ul>
                </div>
            </div>
        );
    }
}