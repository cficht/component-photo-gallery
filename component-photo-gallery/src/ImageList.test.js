import React from 'react';
import { render } from '@testing-library/react';
import images from './data.js';
import ImageList from './ImageList';
import renderer from "react-test-renderer";

test('renders ImageList.js correctly', () => {
  const tree = renderer
  .create(<ImageList images={images} />)
  .toJSON();
  expect(tree).toMatchSnapshot();
});