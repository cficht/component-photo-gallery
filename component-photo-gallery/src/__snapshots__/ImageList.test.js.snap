// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders ImageList.js correctly 1`] = `
<div
  id="list-main"
>
  <div
    id="list-top"
  >
    <select>
      One
    </select>
    <select>
      Two
    </select>
    <p />
  </div>
  <div
    id="image-list"
  >
    <ul>
      <li
        className="list-item"
      >
        <h2>
          UniWhal
        </h2>
        <img
          alt=""
          src="http://3.bp.blogspot.com/_DBYF1AdFaHw/TE-f0cDQ24I/AAAAAAAACZg/l-FdTZ6M7z8/s1600/Unicorn_and_Narwhal_by_dinglehopper.jpg"
        />
        <p>
          A unicorn and a narwhal nuzzling their horns
        </p>
        <h5>
          Keyword: 
          narwhal
        </h5>
        <h5>
          Horns: 
          1
        </h5>
      </li>
      <li
        className="list-item"
      >
        <h2>
          Rhino Family
        </h2>
        <img
          alt=""
          src="https://images.unsplash.com/photo-1512636618879-bbe79107e9e3?ixlib=rb-0.3.5&ixid=eyJhcHBfaWQiOjEyMDd9&s=bd9460ee6d1ddbb6b1ca7be86dfc4590&auto=format&fit=crop&w=1825&q=80"
        />
        <p>
          Mother (or father) rhino with two babies
        </p>
        <h5>
          Keyword: 
          rhino
        </h5>
        <h5>
          Horns: 
          2
        </h5>
      </li>
      <li
        className="list-item"
      >
        <h2>
          Unicorn Head
        </h2>
        <img
          alt=""
          src="https://www.dhresource.com/0x0s/f2-albu-g5-M00-1A-11-rBVaI1hsIIiALxKzAAIHjSU3VkE490.jpg/wholesale-halloween-costume-prop-unicorn.jpg"
        />
        <p>
          Someone wearing a creepy unicorn head mask
        </p>
        <h5>
          Keyword: 
          unicorn
        </h5>
        <h5>
          Horns: 
          1
        </h5>
      </li>
      <li
        className="list-item"
      >
        <h2>
          UniLego
        </h2>
        <img
          alt=""
          src="https://images.unsplash.com/photo-1518946222227-364f22132616?ixlib=rb-0.3.5&ixid=eyJhcHBfaWQiOjEyMDd9&s=4836a6fca62e7dce9324346bacfde085&auto=format&fit=crop&w=2534&q=80"
        />
        <p>
          Lego figurine dressed in a unicorn outfit
        </p>
        <h5>
          Keyword: 
          unilego
        </h5>
        <h5>
          Horns: 
          1
        </h5>
      </li>
      <li
        className="list-item"
      >
        <h2>
          Basically a unicorn
        </h2>
        <img
          alt=""
          src="https://i.pinimg.com/736x/b4/61/06/b46106830b841017ea59870b27ec18dc--narwhals-a-unicorn.jpg"
        />
        <p>
          A narwhal is basically a unicorn after all, right?
        </p>
        <h5>
          Keyword: 
          narwhal
        </h5>
        <h5>
          Horns: 
          1
        </h5>
      </li>
      <li
        className="list-item"
      >
        <h2>
          #truth
        </h2>
        <img
          alt=""
          src="https://i.pinimg.com/originals/16/cf/2a/16cf2a0b3fd51b9bee08bb6296193b75.jpg"
        />
        <p>
          The truth behind narwhals
        </p>
        <h5>
          Keyword: 
          narwhal
        </h5>
        <h5>
          Horns: 
          1
        </h5>
      </li>
      <li
        className="list-item"
      >
        <h2>
          Baby Rhino
        </h2>
        <img
          alt=""
          src="https://secure.img1-ag.wfcdn.com/im/17007094/resize-h800%5Ecompr-r85/3589/35892451/Baby+Rhino+Figurine.jpg"
        />
        <p>
          This is actually a figurine but it looks kinda real
        </p>
        <h5>
          Keyword: 
          rhino
        </h5>
        <h5>
          Horns: 
          2
        </h5>
      </li>
      <li
        className="list-item"
      >
        <h2>
          Cera
        </h2>
        <img
          alt=""
          src="https://vignette.wikia.nocookie.net/landbeforetime/images/c/c3/Cera_infobox.png/revision/latest?cb=20180422005003"
        />
        <p>
          Three horns but still, horns. And who doesn't like The Land Before Time?
        </p>
        <h5>
          Keyword: 
          triceratops
        </h5>
        <h5>
          Horns: 
          3
        </h5>
      </li>
      <li
        className="list-item"
      >
        <h2>
          Narwhal costume
        </h2>
        <img
          alt=""
          src="https://ae01.alicdn.com/kf/HTB18GwSQVXXXXaZaXXXq6xXFXXXh/Animal-Cosplay-Costume-Narwhal-Onesie-Mens-Womens-Cartoon-Whale-Pajamas.jpg"
        />
        <p>
          A woman wearing a blue narwhal costume
        </p>
        <h5>
          Keyword: 
          narwhal
        </h5>
        <h5>
          Horns: 
          1
        </h5>
      </li>
      <li
        className="list-item"
      >
        <h2>
          Rhino costume
        </h2>
        <img
          alt=""
          src="https://www.shopmascot.com/image/cache/mascotnew/new196-800x800.jpg"
        />
        <p>
          Mascots have to get their costumes somewhere
        </p>
        <h5>
          Keyword: 
          rhino
        </h5>
        <h5>
          Horns: 
          2
        </h5>
      </li>
      <li
        className="list-item"
      >
        <h2>
          Believe
        </h2>
        <img
          alt=""
          src="https://www.tinselbox.com/wp-content/uploads/2018/03/I-BELIEVE-IN-UNICORNS-FREE-PRINTABLE-WATERCOLOR-7-of-1.jpg"
        />
        <p>
          I believe in unicorns, do you?
        </p>
        <h5>
          Keyword: 
          unicorn
        </h5>
        <h5>
          Horns: 
          1
        </h5>
      </li>
      <li
        className="list-item"
      >
        <h2>
          Markhor
        </h2>
        <img
          alt=""
          src="https://upload.wikimedia.org/wikipedia/commons/thumb/3/38/Markhor_Schraubenziege_Capra_falconeri_Zoo_Augsburg-02.jpg/220px-Markhor_Schraubenziege_Capra_falconeri_Zoo_Augsburg-02.jpg"
        />
        <p>
          These wild goats eat snakes, then secrete a foam that locals fight over for the antivenom properties -- how cool is that?
        </p>
        <h5>
          Keyword: 
          markhor
        </h5>
        <h5>
          Horns: 
          2
        </h5>
      </li>
      <li
        className="list-item"
      >
        <h2>
          Baby markhor
        </h2>
        <img
          alt=""
          src="http://www.zooborns.com/.a/6a010535647bf3970b0223c84d5959200c-800wi"
        />
        <p>
          Even the babies are adorable
        </p>
        <h5>
          Keyword: 
          markhor
        </h5>
        <h5>
          Horns: 
          2
        </h5>
      </li>
      <li
        className="list-item"
      >
        <h2>
          Mouflon
        </h2>
        <img
          alt=""
          src="https://cdn.modernfarmer.com/wp-content/uploads/2014/08/28476658_9c97f35096_o.jpg"
        />
        <p>
          Those horns though
        </p>
        <h5>
          Keyword: 
          mouflon
        </h5>
        <h5>
          Horns: 
          2
        </h5>
      </li>
      <li
        className="list-item"
      >
        <h2>
          Addax
        </h2>
        <img
          alt=""
          src="https://cdn.modernfarmer.com/wp-content/uploads/2014/08/addax.jpg"
        />
        <p>
          This guy is basically extinct but survives well in captivity, so they're frequently found in zoos
        </p>
        <h5>
          Keyword: 
          addax
        </h5>
        <h5>
          Horns: 
          2
        </h5>
      </li>
      <li
        className="list-item"
      >
        <h2>
          Baby mouflon
        </h2>
        <img
          alt=""
          src="https://cbsnews3.cbsistatic.com/hub/i/r/2013/03/05/5b414225-a645-11e2-a3f0-029118418759/thumbnail/620x350/2d4cf24685b45c22912e64d2004fec8d/Baby_Mouflon_Wild_Sheep.jpg"
        />
        <p>
          The cuteness that is a baby mouflon asleep
        </p>
        <h5>
          Keyword: 
          mouflon
        </h5>
        <h5>
          Horns: 
          2
        </h5>
      </li>
      <li
        className="list-item"
      >
        <h2>
          Happy Jackson's Chameleon
        </h2>
        <img
          alt=""
          src="https://cdn.modernfarmer.com/wp-content/uploads/2014/08/chameleon.jpg"
        />
        <p>
          These are really common in Hawaii
        </p>
        <h5>
          Keyword: 
          chameleon
        </h5>
        <h5>
          Horns: 
          2
        </h5>
      </li>
      <li
        className="list-item"
      >
        <h2>
          Serious Jackson's Chameleon
        </h2>
        <img
          alt=""
          src="https://imgc.allpostersimages.com/img/print/posters/dlillc-jackson-s-chameleon_a-G-13448768-14258384.jpg"
        />
        <p>
          This one is very serious.
        </p>
        <h5>
          Keyword: 
          chameleon
        </h5>
        <h5>
          Horns: 
          3
        </h5>
      </li>
      <li
        className="list-item"
      >
        <h2>
          Horned Lizard
        </h2>
        <img
          alt=""
          src="https://www.nps.gov/band/learn/nature/images/short-horned-lizard-open-mouth-18.jpg?maxwidth=650&autorotate=false"
        />
        <p>
          Fave food: ants
        </p>
        <h5>
          Keyword: 
          lizard
        </h5>
        <h5>
          Horns: 
          100
        </h5>
      </li>
      <li
        className="list-item"
      >
        <h2>
          Smaug
        </h2>
        <img
          alt=""
          src="https://upload.wikimedia.org/wikipedia/commons/thumb/d/df/Smaug_par_David_Demaret.jpg/290px-Smaug_par_David_Demaret.jpg"
        />
        <p>
          Fan illustration of Smaug from 'The Hobbit'
        </p>
        <h5>
          Keyword: 
          dragon
        </h5>
        <h5>
          Horns: 
          100
        </h5>
      </li>
    </ul>
  </div>
</div>
`;
